prime-router/docs/design/proposals/0026-yaml-validation/0026-yaml-validation.md

```diff
@@ -0,0 +1,76 @@
+# `YAML` Validation Proposal
+
+## Background
+
+Our application uses `YAML` in a variety of contexts for configuration. It would speed up development and reduce runtime 
+errors if we were able to validate the `YAML` structure and value types. We want to catch these issues at compile-time during
+development.
+
+## Assumptions
+
+* We will want to continue using `YAML` as our primary configuration language
+* We will want our `YAML` validated every time we either read or update a configuration
+
+## Possible Implementations
+
+### Use Apple's `Pkl` configuration library ([link](https://pkl-lang.org/index.html))
```

Comment:
there are other possibilities

Reply:
None of this is set in stone. I can be convinced if theres a better way!